File: apps/api/app/schemas/activity.py
from datetime import date, datetime, time
from enum import Enum
from typing import Optional

from pydantic import BaseModel


class ActivityFeedItemType(str, Enum):
    MATCH_STARTED = "MATCH_STARTED"
    MATCH_EVENT = "MATCH_EVENT"
    MATCH_FINISHED = "MATCH_FINISHED"
    RECORD_SET = "RECORD_SET"


class ActivityFeedItem(BaseModel):
    id: str
    item_type: ActivityFeedItemType
    created_at: datetime
    title: str
    description: str
    match_id: Optional[int] = None
    event_id: Optional[int] = None
    week_id: Optional[int] = None
    week_number: Optional[int] = None
    sport_id: Optional[int] = None
    sport_name: Optional[str] = None
    modality_id: Optional[int] = None
    modality_name: Optional[str] = None
    event_date: Optional[date] = None
    start_time: Optional[time] = None
    athlete_id: Optional[int] = None
    athlete_name: Optional[str] = None
    delegation_id: Optional[int] = None
    delegation_name: Optional[str] = None
    minute: Optional[int] = None

File: apps/web/src/types/activity.ts
export type ActivityFeedItemType =
  | "MATCH_STARTED"
  | "MATCH_EVENT"
  | "MATCH_FINISHED"
  | "RECORD_SET";

export type ActivityFeedItem = {
  id: string;
  item_type: ActivityFeedItemType;
  created_at: string;
  title: string;
  description: string;
  match_id: number | null;
  event_id: number | null;
  week_id: number | null;
  week_number: number | null;
  sport_id: number | null;
  sport_name: string | null;
  modality_id: number | null;
  modality_name: string | null;
  event_date: string | null;
  start_time: string | null;
  athlete_id: number | null;
  athlete_name: string | null;
  delegation_id: number | null;
  delegation_name: string | null;
  minute: number | null;
};
